sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -378,14 +390,217 @@ private static ExpandoObject BuildBindingMetadataFromAttribute(CustomAttribute a
             bindingDict["Type"] = bindingType;
             bindingDict["Direction"] = GetBindingDirection(attribute);
 
+            // Is string parameter type
+            if (IsStringType(parameterType.FullName))
+            {
+                bindingDict["DataType"] = "String";
+            }
+            // Is binary parameter type
+            else if (IsBinaryType(parameterType.FullName))
+            {
+                bindingDict["DataType"] = "Binary";
+            }
+
             foreach (var property in attribute.GetAllDefinedProperties())
             {
                 bindingDict.Add(property.Key, property.Value);
             }
 
+            // TODO: do not rely on property alone
```

Comment:
In this approach, I do the conversion from public-facing "isBatched" to "cardinality": "many". This is mainly due to the mono.cecil limitations on available information. I don't know if there's a cleaner way without large changes

Reply:
^exactly, and i was also having trouble getting a property that is set by default outside of the constructor, but i might need to double check on that!

Reply:
Flagging this as resolved for now, but as we discussed offline, we'll follow up with enhancements.